packages/teams-js/prepBetaRelease.js

```diff
@@ -102,37 +100,32 @@ function getSpecificVerSuffixNum(versionType, wholeVerNum) {
+ * the number of the new beta version. The new beta version numbers are 0-index based.
+ * @param {string} currBetaVer The version currently tagged beta. (e.g. 2.0.0-beta.1)
+ * @param {string} currPkgJsonVer The version taken from the package.json. (e.g. 2.0.0)
+ * @param {string} nextPrefix The next prefix version that is determined by which is a higher value. (e.g. beta is 2.1.0-beta.8 and package.json is 2.2.0, the next prefix would be 2.2.0)
+ * @returns Just the number of the suffix of the new beta version number. (e.g. return 2 if next beta version is 2.0.0-beta.2)
  */
-function getNewNextDevSuffixNum(currNextDevVer, currPkgJsonVer) {
-  if (currNextDevVer === undefined || currNextDevVer === '') {
+function getNewBetaSuffixNum(currBetaVer, currPkgJsonVer, nextPrefix) {
+  if (currBetaVer === undefined || currBetaVer === '') {
     return 0;
   }
-
-  const betaSuffixNumInCurrNextDev = getSpecificVerSuffixNum('beta', currNextDevVer);
-  const betaSuffixNumInPkgJsonVer = getSpecificVerSuffixNum('beta', currPkgJsonVer);
-  // The new next-dev suffix number will be 0 if the beta suffixes of both the current next-dev version and package.json version are equal.
-  // e.g. package.json version: 2.0.0-beta.1 and current next-dev version: 2.0.0-beta.1-dev.2 means the new next-dev version should be 2.0.0-beta.2-dev.0
-  // and the new next-dev version suffix number is 0.
-  let newNextDevVerSuffixNum = 0;
-
-  // If the beta suffix number in the current next-dev version is already bumped from the beta suffix number of the package.json version,
-  // increment the existing dev suffix number from the next-dev version. e.g. package.json version: 2.0.0-beta.0 and current next-dev version: 2.0.0-beta.1-dev.3
-  // means the new next-dev version will be 2.0.0-beta.1-dev.4 and the new next-dev version suffix number is 4.
-  if (betaSuffixNumInCurrNextDev === betaSuffixNumInPkgJsonVer + 1) {
-    newNextDevVerSuffixNum = getSpecificVerSuffixNum('dev', currNextDevVer) + 1;
-    // If the next-dev beta suffix is not equal to or exactly 1 higher than the package.json beta suffix, throw an error since this does not follow our versioning structure.
-    // e.g. package.json version: 2.0.0-beta.1 and current next-dev version: 2.0.0-beta.0-dev.0 should throw an error.
-  } else if (betaSuffixNumInCurrNextDev !== betaSuffixNumInPkgJsonVer) {
-    throw new Error(
-      `Invalid beta version suffix number ${betaSuffixNumInPkgJsonVer} in package.json version ${currPkgJsonVer}`,
-    );
+  let newBetaSuffixNum = 0;
+
+  //If current beta versioning is higher or equal to currPkgJsonVer
+  if (getPrefix(currBetaVer) === nextPrefix) {
+    const suffixNumInCurrBeta = getSpecificVerSuffixNum('beta', currBetaVer);
+    if (suffixNumInCurrBeta < 0) {
+      throw new Error(`Invalid beta version suffix number ${suffixNumInCurrBeta} in current beta version`);
+    }
+    const suffixNumInPkgJsonVer = getSpecificVerSuffixNum('beta', currPkgJsonVer);
```

Comment:
How would we ever get a "-beta.x" suffix in the package.json version number?

Reply:
I kept this in here since it was part of the logic for next-dev which made me think there are sometimes instances where the package.json version will have a beta version. I didn't think it was likely though due to the `yarn beachball bump` call. I will remove this